pkg/controller/vsphere/actuator.go

```diff
@@ -108,6 +121,9 @@ func (a *Actuator) Exists(ctx context.Context, machine *machinev1.Machine) (bool
 
 func (a *Actuator) Update(ctx context.Context, machine *machinev1.Machine) error {
 	klog.Infof("%s: actuator updating machine", machine.GetName())
+	// Cleanup TaskIDCache so we don't continually grow
+	delete(a.TaskIDCache, machine.Name)
```

Comment:
i just want to understand this better, is the idea here that once we are updating we know we have passed initial need for the creation task id?

Reply:
Once we've hit update() or delete() we know that the task ID was previously successfully recorded, we won't be going down the create() pathway again, so we remove it from the local cache to we don't grow unbounded memory.